client/data/authorizations/actions.ts

```diff
@@ -19,6 +19,51 @@ import {
 import { STORE_NAME } from '../constants';
 import { ApiError } from 'wcpay/types/errors';
 
+const getErrorMessage = ( apiError: {
+	code?: string;
+	message?: string;
+} ): string => {
+	// Map specific error codes to user-friendly messages
+	const errorMessages: Record< string, string > = {
+		wcpay_missing_order: __(
+			'The order could not be found.',
+			'woocommerce-payments'
+		),
+		wcpay_refunded_order_uncapturable: __(
+			'Payment cannot be processed for partially or fully refunded orders.',
+			'woocommerce-payments'
+		),
+		wcpay_intent_order_mismatch: __(
+			'The payment cannot be processed due to a mismatch with order details.',
+			'woocommerce-payments'
+		),
+		wcpay_payment_uncapturable: __(
+			'This payment cannot be processed in its current state.',
+			'woocommerce-payments'
+		),
+		wcpay_capture_error: __(
+			'The payment capture failed to complete.',
+			'woocommerce-payments'
+		),
+		wcpay_cancel_error: __(
+			'The payment cancellation failed to complete.',
+			'woocommerce-payments'
+		),
+		wcpay_server_error: __(
+			'An unexpected error occurred. Please try again later.',
+			'woocommerce-payments'
+		),
+	};
+
+	return (
+		errorMessages[ apiError.code ?? '' ] ??
```

Comment:
> Would the null coalescing operator capture the scenario where the mapping object doesn't have the key in case a new key is added for the code

Good point. Yes, it will be handled falling back to the generic 'Unable to process the payment. Please try again later.'

I added a new test case for this scenario in this commit https://github.com/Automattic/woocommerce-payments/pull/9937/commits/9b8a0d983b52569a4dad541484de9fcf42a07db3 thanks!